src/tests/EndToEndSpec.scala

```diff
@@ -143,13 +143,14 @@ case class EndToEndSpec()(implicit runner: Runner) {
     )
 
   test("allows to fetch using queries") {
-    Dao[TestComplexLongId].query
-      .where(x =>
-        x.innerOpt.exists(_.int >= 2) && x.innerOpt.exists(_.int <= 8)
-      )
-      .orderBy(_.desc(_.innerOpt.map(_.int)))
-      .limit(limit = 2, offset = 1)
-      .find()
+    Dao[TestComplexLongId].all
+      .filter(_.innerOpt.exists(_.int >= 2))
+      .filter(_.innerOpt.exists(_.int <= 8))
+      .sortBy(_.innerOpt.map(_.int))
+      .reverse
+      .drop(1)
+      .take(2)
```

Comment:
Do you think we could also provide a `slice` method that combines `drop` and `take` in a single method?

Reply:
Sure, it shouldn't be a problem.

Reply:
Also I was thinking about supporting cursors, which are way more sufficient. When I was working in my previous project I was adjusting our API to use Elasticsearch API cursors pararell to offset based pagination inside GraphQL. 
In such case new method `after(<cursor>)` may be introduced.

Reply:
That would be useful, I think. Is there generally-accepted behaviour for how cursors work when the data is changing? For example, say you view the first 10 results from a query, and get a cursor to the last item shown. Then, elsewhere, the first five items are deleted, before you request the next ten records after the cursor.

Reply:
In case of offset pagination next request will result with original items 15-25 (since we would request `drop(10).take(10)` while items 1-5 have been deleted)
In case of cursor based implementation we would correctly get results 10-20 since we'll getting next 10 from last cursor which is not relative. In case if entity itself with last cursor would be deleted Datastore can still get correct next results.

If in same example if instead of deletion of elements 1-5 N items would be created in case of offset pagination N items from first result would appear in second result. In case of cursors such behavior should not happen, and result will be equal with the one from first example. 
https://cloud.google.com/datastore/docs/concepts/queries#cursors_and_data_updates